### PR TITLE
[ClangFormat] Fix indent in child lines within a macro argument.

### PR DESCRIPTION
**Pull request by r4nt**
When reconstructing lines from a macro expansion, make sure that lines at different levels in the expanded code get indented correctly as part of the macro argument.

### CODE REVIEW

File: clang/unittests/Format/FormatTestMacroExpansion.cpp [hunk 1]
```diff
@@ -287,6 +287,21 @@ TEST_F(FormatTestMacroExpansion,
...
+  Style.Macros.push_back("MACRO(a, b)=a=(b)");
+  verifyFormat("void f() {\n"
+               "  MACRO(a b, call([] {\n"
+               "          if (expr) {\n"
+               "            indent();\n"
+               "          }\n"
+               "        }));\n"
+               "}"
+
+               ,
```
Comment (owenca) [inline clang/unittests/Format/FormatTestMacroExpansion.cpp:301]:
```suggestion
               "}",
```

File: clang/lib/Format/UnwrappedLineParser.h [hunk 2]
```diff
@@ -420,6 +420,8 @@ struct UnwrappedLineNode {
   SmallVector<UnwrappedLine, 0> Children;
 };
 
+std::ostream &operator<<(std::ostream &Stream, const UnwrappedLine &Line);
+
```
Comment (owenca) [inline clang/lib/Format/UnwrappedLineParser.h:424]: Delete.

File: clang/lib/Format/UnwrappedLineParser.cpp [hunk 3]
```diff
@@ -90,6 +90,12 @@ class ScopedDeclarationState {
 
 } // end anonymous namespace
 
+std::ostream &operator<<(std::ostream &Stream, const UnwrappedLine &Line) {
+  llvm::raw_os_ostream OS(Stream);
+  printLine(OS, Line);
+  return Stream;
+}
+
```
Comment (owenca) [inline clang/lib/Format/UnwrappedLineParser.cpp:98]: Delete.

File: clang/lib/Format/MacroCallReconstructor.cpp [hunk 4]
```diff
@@ -19,6 +19,7 @@
 #include "clang/Basic/TokenKinds.h"
 #include "llvm/ADT/DenseSet.h"
 #include "llvm/Support/Debug.h"
+#include <algorithm>
```
Comment (owenca) [inline clang/lib/Format/MacroCallReconstructor.cpp:22]: Not needed explicitly.

File: clang/lib/Format/UnwrappedLineParser.h (hunk 2 above)
Comment (owenca) [reply clang/lib/Format/UnwrappedLineParser.h:424]: Ah, thanks for the explanation.

Comment (r4nt) [reply clang/lib/Format/UnwrappedLineParser.h:424]: I can move the operator<< to the test header, but I'd then at least need some public method to do the printing.

**Pull request merged**